### ISSUE TITLE
Error: This session is not valid for use with the REST API

### ISSUE THREAD

**Issue by boukeas** [simple-salesforce/simple-salesforce#422]
Hello. Using `simple_salesforce` ver. 1.10.1 and the code below, I am able to login to a sandbox:
```
from simple_salesforce import Salesforce
sf = Salesforce(
    username='...', 
    password='...', 
    security_token='...',
    domain='test')
```
I know the login has been successful by checking the login history and account activity in Salesforce. However, I am unable to perform a `sf.search` or `sf.query`. They fail with the following exception/error message:
```
simple_salesforce.exceptions.SalesforceExpiredSession: Expired session for https://.../services/data/v42.0/search/?q=.... 
Response content: [{'message': 'This session is not valid for use with the REST API', 'errorCode': 'INVALID_SESSION_ID'}]
```
My Salesforce administrator is certain that API access is enabled. Is there anything else I should be doing while logging in or querying?

**Comment by namper** [comment]
I think you should also include _instance_url_

```
from simple_salesforce import Salesforce
sf = Salesforce(
    username='...', 
    password='...', 
    security_token='...',
    instance_url='https://eu10.salesforce.com',
    domain='test'
)
```

**Comment by boukeas** [comment]
Thank you @namper but that didn't make a difference. Like I said, the login seems to be successful: through the Salesforce web interface, I can see the login and the new session. It's the queries after the login that fail.

**Comment by boukeas** [comment]
This was a permissions issue on the Salesforce side, so nothing to do with simple-salesforce. Thank you.

**Comment by 3supower** [comment]
@boukeas So, what was the solution from the Salesforce side? Was that something to do with two factor authentication in session settings or something?

**Comment by amckeachie-silius** [comment]
We had the same issue.  We had to update the 'Session Security Level Required at Login' profile setting for the profile of the user that we are connecting to Salesforce with from simple-salesforce.  When this setting is set to 'High Assurance' we get the 'This session is not valid for use with the REST API' error.  When we set it to '--None--', we are able to both login to and query Salesforce successfully.

Setup -> Profile -> Session Security Level Required at Login -> --None--
